client/src/ui/organisms/article-actions/language-menu/index.tsx

```diff
@@ -127,3 +126,37 @@ function LanguageMenuItem({
     </a>
   );
 }
+
+function LocaleRedirectSetting() {
+  const gleanClick = useGleanClick();
+  const locale = useLocale();
+  const [value, setValue] = useState(false);
+
+  useEffect(() => {
+    setValue(!!getCookieValue(PREFERRED_LOCALE_COOKIE_NAME));
+  }, [setValue]);
```

Comment:
I don't think it's necessary to include `setValue` as a dependency here, see what the linter says if you remove:
```suggestion
  }, []);
```
The risk of adding it if it isn't necessary is it makes the intent of the code a bit unclear: originally I was thinking "wait if `setValue` is a dep, and we set value, won't this be an infinite loop" - obviously not, since `setValue` itself doesn't change, but `[]` immediately says to me: "this runs on component creation and destruction on the client".